app/controllers/users_controller.rb

```diff
@@ -2,24 +2,28 @@ class UsersController < ApplicationController
   include Accessible
 
   def new
+    @person = Person.new
+    @person.personable = User.new
   end
 
   def create
-    ActiveRecord::Base.transaction do
-      @person = Person.new(email: user_params[:email])
-      @user = User.new(password: user_params[:password], password_confirmation: user_params[:password])
-      @person.personable = @user
-
-      respond_to do |format|
-        if @person.save
-          reset_session
-          session[:current_user_id] = @user.id
-          format.html { redirect_to dashboard_path, notice: "Account was successfully created." }
-          format.json { render :show, status: :created, location: @user }
-        else
-          format.html { render :new }
-          format.json { render json: @user.errors, status: :unprocessable_entity }
-        end
+    @person = Person.new
+    @person.personable_type = "User"
+    @person.update person_params
+
```

Comment:
But it actually feels backwards that the `def new` and the `def create` are treating @person as the object we're dealing with, when this is the users_controller. Can't we do the reverse and always initialize and create User objects? I'm pretty sure it would work to:

1. Change `def new` to do `@user = User.new` by adding this to User.rb.

```
after_initialize :set_default_person

private

def set_default_person
  self.person ||= Person.new
end
```

I think this is actually better than what I proposed in the comment above, because a User *always* has a Person but a Person does not always have a User.

2. Change this create to simply be `@user = User.new ...` and within User.rb you can then add:

```
  has_one :person, as: :personable, touch: true, autosave: true
  accepts_nested_attributes_for :person
```

Reply:
I learned some things today, and re-learned some other things, haha. Good call -- I didn't think you could do nested attributes across `has_` associations, but they've done a good job of making the api more robust than I thought.

Reply:
I wasn't sure either! ChatGPT for the win. This is the kind of "architecture conversation" I have with ChatGPT so often these days. It turns out that I have a friend named Samantha who knows the rails API *really* well. :) This might be interesting for you to see or it might be old hat, by now:

<img width="741" alt="image" src="https://github.com/hostedgpt/hostedgpt/assets/35061/2d4cc1aa-4c71-40ef-8919-8e902b7e8f3d">